src/Tooltip2.tsx

```diff
@@ -0,0 +1,292 @@
+        }),
+    }),
+  },
+
+  '&::before, & > span': {
+    ...(open && {
+      display: 'inline-block',
+      textDecoration: 'none',
+      animationName: 'tooltip-appear',
+      animationDuration: '0.1s',
+      animationFillMode: 'forwards',
+      animationTimingFunction: 'ease-in',
+      animationDelay: noDelay ? '0s' : '0.4s',
+    }),
+  },
+})
+
+const Tooltip2: React.FC<React.PropsWithChildren<Tooltip2Props>> = ({
+  direction = 'n',
+  text,
+  noDelay,
+  align,
+  wrap,
+  sx = {},
+  children,
+  type = 'label',
+  ...props
+}) => {
+  const id = useId()
+  const tooltipRef = useRef<HTMLDivElement>(null)
+  const child = Children.only(children) // make sure there is only one child
+  const [open, setOpen] = useState(false)
+
+  // we need this check for every render
+  useEffect(() => {
+    if (tooltipRef.current) {
+      const childNode = tooltipRef.current.children[0] // For now, I assume it has one node but that is not true
+      if (!isFocusable(childNode as HTMLElement)) {
+        throw new Error('Tooltip2: The child element must be focusable')
+      }
+    }
+  })
+  const triggerProps = {
+    // if it is a type description, we use tooltip to describe the trigger
+    'aria-describedby': type === 'description' ? id : undefined,
+    // If it is a type description, we should keep the aria label if it exists, otherwise we remove it because we will use aria-labelledby
+    'aria-label': type === 'description' ? (children as React.ReactElement).props['aria-label'] : undefined,
+    //   If it is a label type, we use tooltip to label the trigger
+    'aria-labelledby': type === 'label' ? id : undefined,
+  }
+
+  // Only need tooltip role if the tooltip is a description for supplementary information
+  const role = type === 'description' ? 'tooltip' : undefined
+  // aria-hidden true only if the tooltip is a label type
+  const ariaHidden = type === 'label' ? true : undefined
+
+  return (
+    <>
+      <style>
+        {`@keyframes tooltip-appear {
+            from {
+              opacity: 0;
+            }
+        
+            to {
+              opacity: 1;
+            }
+          }`}
+      </style>
+      <Box
+        ref={tooltipRef}
+        sx={merge<BetterSystemStyleObject>(tooltipStyle({direction, noDelay, align, wrap, open}), sx)}
+        {...props}
+      >
+        {React.cloneElement(child as React.ReactElement<TriggerPropsType>, {
+          ...triggerProps,
+          // Optimise this?
+          onFocus: () => {
+            setOpen(true)
+          },
+          // onBlur: () => {
+          //   setOpen(false)
+          // },
+          onMouseEnter: () => {
+            setOpen(true)
+          },
+          // onMouseLeave: () => {
+          //   setOpen(false)
+          // },
+          onKeyDown: (e: React.KeyboardEvent) => {
+            if (open && e.key === 'Escape') {
+              e.stopPropagation()
+              setOpen(false)
+            }
+          },
```

Comment:
One idea, we should make sure to compose any handlers we had with what the `child` may have.

In other words, if someone writes:

```js
<Tooltip>
  <Button onKeyDown={() => { /* ... */ }} />
</Tooltip>
```

We should make sure not to override their `onKeyDown`

Reply:
oh true! 🙈 I'll do that